src/libpmem/pmem.c

```diff
@@ -611,8 +664,154 @@ memmove_nodrain_movnt(void *pmemdest, const void *src, size_t len)
 	/* way too chatty for LOG level 3 */
 	LOG(15, "pmemdest %p src %p len %zu", pmemdest, src, len);
 
-	/* XXX stub version */
-	return memmove(pmemdest, src, len);
+	__m128i xmm0, xmm1, xmm2, xmm3, xmm4, xmm5, xmm6, xmm7;
+	size_t i;
+	__m128i *d;
+	__m128i *s;
+	void *dest1 = pmemdest;
+	size_t cnt;
+
+	if (len == 0)
+		return NULL;
+
+	if (src == pmemdest)
+		return pmemdest;
+
+	if (src > pmemdest) {
```

Comment:
Ok. Will change. The way I did it here works for all cases but I can see your point.